### ISSUE TITLE
Arduino Uno PD1 Problem

### ISSUE THREAD

**Issue by CloudVIII** [lcgamboa/picsimlab#25]
Hi, wrote code with asm that uses pin PD1 for an ATmega328P and can't use it properly in PICSimLab. Verified it works in the Atmel simulator and that UART communication it's disabled.

**Comment by lcgamboa** [comment]
Apparently the simavr initializes the UCSR0B register with the value 0x08 (instead of 0x00), enabling the TX pin of the serial and disabling the PD1. Assigning the value 0x00 to UCSR0B at the beginning of the atmega code solves the operating problem. I will try to find the exact source of the problem.

**Comment by lcgamboa** [comment]
Confirmed, the issue is in the simavr (used in picsimlab background to simulate avr mcus):
https://github.com/buserror/simavr/blob/0e03bc6f48b078606b9a970594ee0ef82ffa0117/simavr/sim/avr_uart.c#L466
The simavr force the enabling of TX pin for debug purpose.

**Comment by CloudVIII** [comment]
Sadly, already tried that in the code. Still the same problem.

**Comment by lcgamboa** [comment]
I have tested with this c code:

```
#define F_CPU 2000000L
#include <avr/io.h>
#include <util/delay.h>

int main()
{
    unsigned char val=0x01;
    DDRB=0xFF;
    DDRC=0xFF;
    DDRD=0xFF;
    UCSR0B=0x00; //need to work in simavr
    PORTD|=0x02;  //need to work in simavr
    PORTD&=~0x02;  //need to work in simavr
    while(1)
    {
        PORTB=val;
        PORTC=val;
        PORTD=val;
        val<<=1;
        if(!val)val=0x01;
        _delay_ms(500);
    }
}
```
Can you send me your sample code with problem?

**Comment by lcgamboa** [comment]
Simavr txpin wrong state on reset fixed in commit ddc38c6

**Comment by CloudVIII** [comment]
It works after the last commit.
